parity-clib/parity.h

```diff
@@ -88,92 +120,112 @@ int parity_config_from_cli(char const* const* args, size_t const* arg_lens, size
 ///		return 1;
 /// }
 /// char[] logger_mode = "rpc=trace";
-/// parity_set_logger(logger_mode, strlen(logger_mode), nullptr, 0, &cfg.logger);
+/// parity_set_logger(logger_mode, strlen(logger_mode), nullptr, 0,
+/// &cfg.logger);
 /// ```
 ///
-int parity_set_logger(const char* log_mode, size_t log_mode_len, const char* log_file, size_t log_file_len, void** logger);
+int parity_set_logger(const char *log_mode, size_t log_mode_len,
+                      const char *log_file, size_t log_file_len,
+                      struct parity_logger **logger);
 
 /// Destroys a configuration object created earlier.
 ///
-/// **Important**: You probably don't need to call this function. Calling `parity_start` destroys
-/// 				the configuration object as well (even on failure).
-void parity_config_destroy(void* cfg);
+/// **Important**: You probably don't need to call this function. Calling
+/// `parity_start` destroys the configuration object as well (even on failure).
+///
+/// It is safe to pass NULL here, in which case this function has no effect.
+void parity_config_destroy(struct parity_config *cfg);
 
-/// Starts the parity client in background threads. Returns a pointer to a struct that represents
-/// the running client. Can also return NULL if the execution completes instantly.
+/// Starts the parity client in a background thread. Returns a pointer to a
+/// struct that represents the running client. Can also return NULL if the
+/// execution completes instantly.
```

Comment:
When would this happen? Like, is `NULL` an error condition or can it be legitimate?